app/services/github/contributions/wrappers/pull_request.rb

```diff
@@ -16,12 +19,78 @@ def repository_name
           @repository_name ||= pull_request.html_url.split("/")[-4..-3].join("/")
         end
 
+        def repository_owner
+          @repository_owner ||= pull_request.html_url.split("/")[-4]
+        end
+
+        def pull_request_number
+          @pull_request_number ||= pull_request.html_url.split("/")[-1]
+        end
+
+        def pull_request_commits
+          @pull_request_commits ||= client.pull_requests.commits(
+            user: repository_owner, repo: repository_name.split("/")[-1], number: pull_request_number)
+        end
+
+        def get_co_authors_data_by_commit_messages
+          co_authors_data = pull_request_commits.filter_map do |commit|
+            commit_message = commit.dig("commit", "message")
+
+            data = nil
+
+            if commit_message.match?(CO_AUTHOR_REGEX)
+              data = commit_message.scan(CO_AUTHOR_REGEX).flatten
+              data.first.strip!
+            end
+
+            data
+          end
+          Set.new(co_authors_data)
+        end
+        
+        def get_co_authors_data_by_committers
+          co_authors_data = pull_request_commits.map do |commit|
+            login = commit.dig("committer", "login")
+
+            login unless pull_request.user.login == login
```

Comment:
I did not get the need of this

Reply:
This covers the second scenario in which a co author is not defined on commit's message. I consider every committer that's not the PR's author as a co author too.